src/Compilers/CSharp/Test/Semantic/Semantics/IteratorTests.cs

```diff
@@ -106,6 +106,178 @@ IEnumerable<int> I()
                 );
         }
 
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/72443")]
+        public void YieldInLock_Async()
+        {
+            var source = """
+                using System;
+                using System.Collections.Generic;
+                using System.Threading;
+                using System.Threading.Tasks;
+
+                try
+                {
+                    await new C().ProcessValueAsync();
+                }
+                catch (SynchronizationLockException e)
+                {
+                    Console.WriteLine(e.Message);
```

Comment:
>Console.WriteLine(e.Message); [](http://example.com/codeflow?start=20&length=29)

Was the intent to execute the test code? If not, perhaps remove the caller.

Reply:
Originally the intent was to execute the test code; but that was removed in https://github.com/dotnet/roslyn/pull/72755#discussion_r1546414147. I will remove the caller, thanks.